code/game/machinery/telecomms/machines/hub.dm

```diff
@@ -34,6 +34,36 @@
 
 	use_power(idle_power_usage)
 
+/obj/machinery/telecomms/hub/update_power()
+	var/old_on = on
+	if(toggled)
+		if(machine_stat & (BROKEN|NOPOWER|EMPED)) // if powered, on. if not powered, off. if too damaged, off
+			on = FALSE
+			soundloop.stop()
+		else
+			on = TRUE
+			soundloop.start()
+	else
```

Comment:
you should instead to `if (toggled && (machine_stat & (BROKEN|NOPOWER|EMPED)))` then you only need one else

Reply:
Could you elaborate on this point for a mid-skill coder like myself?

Reply:
you check toggled. if its not toggled, you turn it off
then you check the machine's state. if it's not right, you turn it off

you can check both of these at once